hoodie-utilities/src/main/java/com/uber/hoodie/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -204,8 +204,10 @@ public KafkaOffsetGen(TypedProperties props) {
 
     // Determine the offset ranges to read from
     HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> fromOffsets;
+    HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> checkpointOffsets;
     if (lastCheckpointStr.isPresent()) {
-      fromOffsets = CheckpointUtils.strToOffsets(lastCheckpointStr.get());
+      checkpointOffsets = CheckpointUtils.strToOffsets(lastCheckpointStr.get());
+      fromOffsets = checkupValidOffsets(cluster, checkpointOffsets, topicPartitions);
```

Comment:
could we do this by upon exception alone? without incurring this overhead in every time?

Reply:
Every time we get the checkpoint offset, we need to verify that it is valid. Do you have a better way ?

Reply:
I was thinking if there was a specific exception we can catch.. then we log that and fetch the earliest offsets and continue.. wdyt?

Reply:
I agree with you, and will change my code.

Reply:
the exception is thrown in KafkaUtils 
```
 /** Make sure offsets are available in kafka, or throw an exception */
  private def checkOffsets(
      kc: KafkaCluster,
      offsetRanges: Array[OffsetRange]): Unit = {
    val topics = offsetRanges.map(_.topicAndPartition).toSet
    val result = for {
      low <- kc.getEarliestLeaderOffsets(topics).right
      high <- kc.getLatestLeaderOffsets(topics).right
    } yield {
      offsetRanges.filterNot { o =>
        low(o.topicAndPartition).offset <= o.fromOffset &&
        o.untilOffset <= high(o.topicAndPartition).offset
      }
    }
    val badRanges = KafkaCluster.checkErrors(result)
    if (!badRanges.isEmpty) {
      ("Offsets not available on leader: " + badRanges.mkString(","))
    }
  }
``` 
which is called in AvroKafkaSource class and cannot get the exception